gframe/data_handler.cpp

```diff
@@ -49,11 +49,24 @@ void DataHandler::LoadPicUrls() {
 					auto type = obj["type"].get<std::string>();
 					if(obj["url"].get<std::string>() == "default") {
 						if(type == "pic") {
+							if (configs->toggle_hd_card_pics) {
+#ifdef DEFAULT_HD_PIC_URL
+								imageDownloader->AddDownloadResource({ DEFAULT_HD_PIC_URL, ImageDownloader::ART });
+#else
```

Comment:
#elseif is a thing, also, this isn't a proper approach, it should be implemented by adding a new field to the json file, then teh client will choose the sources that have that field set/unset depending on the user setting. with this, the user will be stuck with only the hardcoded hq sources and also this change will only apply oncve the game is restarted, not at runtime

Reply:
Those are both details I had considered while making this.
1) New field on JSON file: Is it really something we have to implement? Having a new field for solely the distinction of HD sources of card art (which would be specific to card

Reply:
The purpose of the json itself is to be able to add new fields more easily, and the client aims into customizability, so yes, a new field is needed, because in that way you could make your custom packs and provide proper image sources as well, that's the primary goal of all this customizability.
For the runtime behaviour, it would really be a matter of just checking if the setting is toggled, and switch to the hq sources accordingly, handling previously downloaded images shouldn't be our problem.
For the "reset", do you mean when a pic is downloaded, then it won't be downloaded again if missing? That's still intended behaviour, the client shouldn't download a pic indefinitely if it's for example deleted by the antivirus, that would only waste resources, and if it weirdly disappeared, then it would be a bigger issue than the handling in the client.

Reply:
The behavior you just mentioned has been implemented in the commits below, thanks for the feedback.